app/models/search_builder.rb

```diff
@@ -141,6 +141,12 @@ def modify_combo_param_with_absent_q(solr_parameters)
   # a sort, we sort by id to provide stable deterministic ordering.
   def override_sort_when_q_is_empty(solr_parameters)
     blacklight_sort = blacklight_params[:sort]
+    if blacklight_params[:search_field] == 'subject_correlation'
+      solr_parameters[:presentation_domain] = '{!filters param=$fq excludeTags=cluster,no_correlation}'
+      solr_parameters[:sort] = ''
+      solr_parameters[:rows] = 0
+      return
+    end
```

Comment:
it seems like this should be it's own method in `search_builder` rather than piggybacking on this method

Reply:
I guess that would be more idiomatic, and could be made to work in this case. FWIW I'm not thrilled with the fact that all these methods are manipulating the same state/variables without a clear way to conditionally prioritize one decision over another. The order of the application of these methods is hard-coded (order added to `:default_processor_chain`) -- so the decisions of one component of the chain can be totally overwritten by the decisions of a downstream component. I think the main reason I put these both in the same method was because it's manipulating the same variables, and if separated into another method, that other method would have to basically check what the _other_ method was doing to decide how it should proceed itself, which felt wrong to me.
Would this concern be addressed by renaming the top-level method (added to `:default_processor_chain` to be more general (e.g., `filter_sort_and_rows` instead of `override_sort_when_q_is_empty`), and optionally refactor that method to call other methods?

Reply:
i think a renaming of this method will be sufficient for now